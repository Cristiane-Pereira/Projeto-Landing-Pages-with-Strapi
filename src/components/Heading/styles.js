import styled, { css } from 'styled-components';

export const titleSize = {
  small: (theme) => css`
    font-size: ${theme.fonts.size.medium};
  `,
  medium: (theme) => css`
    font-size: ${theme.fonts.size.large};
  `,
  large: (theme) => css`
    font-size: ${theme.fonts.size.xlarge};
  `,
  huge: (theme) => css`
    font-size: ${theme.fonts.size.xhuge};
  `,
};

export const mediaQuery = (theme) => css`
  @media ${theme.media.mobile} {
    font-size: ${theme.fonts.size.large};
  }
`;

export const titleCase = (uppercase) => css`
  text-transform: ${uppercase ? 'uppercase' : 'none'};
`;

export const Title = styled.h1`
  ${({ theme, themeDark, size, uppercase }) => css`
    color: ${themeDark ? theme.colors.white : theme.colors.dark};
    ${titleSize[size](theme)}
    ${titleCase(uppercase)}
    ${mediaQuery(theme)}
  `}
`;
