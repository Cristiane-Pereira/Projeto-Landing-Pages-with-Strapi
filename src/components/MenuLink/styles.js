import styled, { css } from 'styled-components';

export const Container = styled.a`
  ${({ theme }) => css`
    display: block;
    text-decoration: none;
    font-size: ${theme.fonts.size.small};
    padding: 0px 16px;
    color: ${theme.colors.backgroundPrimary};
    position: relative;
    &::after {
      content: '';
      position: absolute;
      top: 2rem;
      bottom: 0.76rem;
      left: 50%;
      width: 0;
      height: 0.2rem;
      background: ${theme.colors.backgroundSecond};
      transition: all 300ms ease-in-out;
    }
    &:hover::after {
      left: 25%;
      width: 50%;
    }
    @media ${theme.media.mobile} {
      margin-bottom: 1.6rem !important;
    }
  `}
`;
