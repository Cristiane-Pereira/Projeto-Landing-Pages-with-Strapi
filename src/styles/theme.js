const theme = {
  colors: {
    backgroundPrimary: 'purple',
    backgroundSecond: 'blue',
    white: '#fff',
  },
  fonts: {},
  spacing: {},
};

export default theme;
