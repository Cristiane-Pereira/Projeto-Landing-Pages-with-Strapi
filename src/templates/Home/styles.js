import styled, { css } from 'styled-components';

// eslint-disable-next-line prettier/prettier
export const Container = styled.div`
  ${(props) => css`
    background: ${props.background};
  `}
  width: 100vw;
  height: 100vh;
  display: flex;
  align-items: center;
  justify-content: center;
`;
