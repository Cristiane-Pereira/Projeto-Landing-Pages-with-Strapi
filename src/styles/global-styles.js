import { createGlobalStyle, css } from 'styled-components';
import theme from './theme';

export const GlobalStyles = createGlobalStyle`
* {
 margin: 0;
 padding: 0;
 box-sizing: border-box;
}
  html {
    font-size: 62.5%;
  }
  body {
    font-size: 1.6rem;
    font-family: ${({ theme }) => theme.fonts.family.default};
      /* ${() => css`
        background: ${theme.colors.backgroundPrimary};
      `} */
  }
  h1, h2, h3, h5, h5, h6 {
    font-family: ${({ theme }) => theme.fonts.family.secondary};
  }
`;
