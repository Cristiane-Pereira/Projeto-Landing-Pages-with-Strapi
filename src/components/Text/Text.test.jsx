import { screen } from '@testing-library/react';
import { ThemeProvider } from 'styled-components';
import { Text } from '.';
import { renderTheme } from '../../styles/render-theme';
import { theme } from '../../styles/theme';

describe('<Text />', () => {
  it('should render a text', () => {
    renderTheme(
      <Text>
        Lorem ipsum dolor, sit amet consectetur adipisicing elit. Laboriosam
        quam tenetur alias animi in atque dolorem! Hic voluptatibus inventore
        quas reiciendis accusantium facere, similique nobis impedit.Excepturi
        itaque doloribus facere!
      </Text>,
    );
    expect(
      screen.getByText(`Lorem ipsum dolor, sit amet consectetur adipisicing elit. Laboriosam
        quam tenetur alias animi in atque dolorem! Hic voluptatibus inventore
        quas reiciendis accusantium facere, similique nobis impedit.Excepturi
        itaque doloribus facere!`),
    ).toBeInTheDocument();
  });

  it('should match snapshot', () => {
    const { container } = renderTheme(
      <Text>
        Lorem ipsum dolor, sit amet consectetur adipisicing elit. Laboriosam
        quam tenetur alias animi in atque dolorem! Hic voluptatibus inventore
        quas reiciendis accusantium facere, similique nobis impedit.Excepturi
        itaque doloribus facere!
      </Text>,
    );
    expect(container.firstChild).toMatchSnapshot();
  });
});
