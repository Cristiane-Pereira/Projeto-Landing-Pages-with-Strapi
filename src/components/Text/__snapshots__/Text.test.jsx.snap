// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Text /> should match snapshot 1`] = `
.c0 {
  font-size: 2.4rem;
}

<p
  class="c0"
>
  Lorem ipsum dolor, sit amet consectetur adipisicing elit. Laboriosam quam tenetur alias animi in atque dolorem! Hic voluptatibus inventore quas reiciendis accusantium facere, similique nobis impedit.Excepturi itaque doloribus facere!
</p>
`;
